### PR TITLE
Remove react-autosize-textarea

### PR DESCRIPTION
**Pull request by notbakaneko**
`react-autosize-textarea` hasn't been updated in a while and doesn't have current versions of React in its dependencies.
It's just wrapping `autosize`, anyway, so we might as well use `autosize` directly - the component is just a convenience wrapper to attach and detach `autosize` and doesn't use `forwardRef`, making the implementation more straight-forward.

### CODE REVIEW

File: resources/js/components/textarea-autosize.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export default class TextareaAutosize extends React.Component<Props> {
+  static readonly defaultProps = {
+    async: false,
+    rows: 1,
+  };
+
+  private lineHeight?: number;
+  private ref = this.props.innerRef ?? React.createRef<HTMLTextAreaElement>();
```
Comment (nanaya) [inline resources/js/components/textarea-autosize.tsx:20]: should this be a getter instead so the value doesn't potentially get outdated when the props change? (although probably practically impossible? also there are a bunch of other ref-related things that's being set once in componentDidMount and assumed being the same in componentDidUpdate)

File: resources/js/chat/input-box.tsx [hunk 2]
```diff
@@ -109,10 +109,10 @@ export default class InputBox extends React.Component<Props> {
     return (
       <div className='chat-input'>
         <TextareaAutosize
-          ref={this.inputBoxRef}
           autoComplete='off'
           className={classWithModifiers('chat-input__box', { disabled: this.inputDisabled })}
           disabled={this.inputDisabled}
+          innerRef={this.inputBoxRef}
           maxLength={channel?.messageLengthLimit ?? maxMessageLength}
           maxRows={channel?.type === 'ANNOUNCE' ? 10 : 3}
```
Comment (nanaya) [inline resources/js/chat/input-box.tsx:117]: looks like the maximum rows option doesn't get applied correctly

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/js/components/textarea-autosize.tsx (hunk 1 above)
Comment (notbakaneko) [reply resources/js/components/textarea-autosize.tsx:23]: Refs aren't managed by React other than committing a value to it after render, so messing with a ref wouldn't normally be triggering renders anyway

File: resources/js/components/textarea-autosize.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    autosize.destroy(this.ref.current);
+  }
+
+  render() {
+    const { async, innerRef, maxRows, style, ...otherProps } = this.props;
+
+    const maxHeight = this.maxHeight;
+
+    return (
+      <textarea
```
Comment (nanaya) [inline resources/js/components/textarea-autosize.tsx:72]: not sure exactly where but the styling seems to be slightly wrong

old


this pr


note the wrong background colour for the padding

<details>
<summary>it's slightly more <em>interesting</em> in chrome</summary>



</details>

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/js/components/textarea-autosize.tsx (hunk 3 above)
Comment (notbakaneko) [reply resources/js/components/textarea-autosize.tsx:72]: The old round uses `Math.round` before multiplying the rows, this one applies `Math.ciel` after multiplying; current `line-height` is `17.5px`

Comment (notbakaneko) [reply resources/js/components/textarea-autosize.tsx:72]: white box appears to be a change in `autosize` between 4 and the current version where previously it applied `overflow: hidden scroll` but not it doesn't 🤔

**Review by nanaya — APPROVED** [review]

**Pull request merged**